sympy/calculus/range_func.py

```diff
@@ -0,0 +1,170 @@
+from sympy import sympify, diff, limit, S, oo
+from sympy.core import Expr
+from sympy.core.numbers import Number
+from sympy.calculus.singularities import singularities
+from sympy.sets.sets import Interval, Intersection, FiniteSet, Union, Complement, Set
+from sympy.simplify import simplify
+from sympy.solvers.solveset import solveset_real
+
+
+def range_func(func, set_value):
```

Comment:
I would like you to explicitly take the involved variables as a tuple rather than determining them using `func.free_symbols`. As a general principle `explicit is better than implicit` plus in this case it will help us return paratmetric `codomain` say for things like the general quadratic equation.